packages/core/helper-plugin/lib/src/hooks/useFetchClient/tests/useFetchClient.test.js

```diff
@@ -1,21 +1,59 @@
-import { renderHook } from '@testing-library/react-hooks';
+import React, { useEffect } from 'react';
+import { renderHook, act } from '@testing-library/react-hooks';
+import { useFetchClient } from '@strapi/helper-plugin';
 
-import useFetchClient from '../index';
+jest.mock('@strapi/helper-plugin', () => ({
+  ...jest.requireActual('@strapi/helper-plugin'),
+  useFetchClient: jest.fn().mockReturnValue({
+    get: jest.fn().mockResolvedValue({
+      data: {
+        results: [
+          { id: 2, name: 'newest', publishedAt: null },
+          { id: 1, name: 'oldest', publishedAt: null },
+        ],
+        pagination: { page: 1, pageCount: 10 },
+      },
+    }),
+    post: jest.fn(),
+    put: jest.fn(),
+    del: jest.fn(),
+  }),
+}));
 
-function setup() {
-  return renderHook(() => useFetchClient());
+const TestComponent = (props) => {
+  const { get } = useFetchClient();
+  useEffect(() => {
+    get('/foo');
+  }, [get]);
+
+  // eslint-disable-next-line react/prop-types
+  return <div {...props}>{props.children}</div>;
+};
+
+function setup(props) {
+  return new Promise((resolve) => {
+    act(() => {
+      resolve(
+        renderHook(() => useFetchClient(), {
+          wrapper: ({ children }) => <TestComponent {...props}>{children}</TestComponent>,
+        })
+      );
+    });
+  });
 }
 
 describe('useFetchClient', () => {
   afterEach(() => {
     jest.clearAllMocks();
   });
-  it('Should contain the get, del, put, post methods', async () => {
+  it('Should contain the get, del, put, post methods and call once the GET method', async () => {
     const { result } = await setup();
 
     expect(result.current).toHaveProperty('get');
     expect(result.current).toHaveProperty('post');
     expect(result.current).toHaveProperty('put');
     expect(result.current).toHaveProperty('del');
+
+    expect(result.current.get).toHaveBeenCalledTimes(1);
```

Comment:
Mh, I think if you want to make that test useful you'd have to re-render and confirm it still has only be called once, because right now this is expected no?

Reply:
yes sorry, I push the first draft of the test, now I added the rerendering and if the get is called once even after the rerendering

Reply:
I don't think this test is reliable or correct. You're testing against the `renderHook` instance of `useFetchClient`, you're not testing against the instance of `TestComponent`. Because `wrapper` just wraps adds a component around the internal test component of `@testing-library/react-hooks`...

You should just be using `@testing-library/react`'s render method to render the component with the hook inside and assert against that by spying/mocking the hook.